include/envoy/safe_init/BUILD

```diff
@@ -0,0 +1,31 @@
+licenses(["notice"])  # Apache 2
```

Comment:
At a higher level question, will this be `include/envoy/init` at the very end or we keep the `safe_init` directory? Everything currently on Init::{Manager,Target} will be transformed to SafeInit::{Manager,Target} right?

Reply:
That was going to be one of my high level comments. I don't think we need "safe" in the directory or classes here. As long as we have a plan to winding up there eventually I'm OK with whatever.

Reply:
Yeah, I figure there's a few discrete steps to this (just to make the PRs digestible):

* Get the `SafeInit` implementation up in this PR.
* Migrate everything currently on `Init` to `SafeInit`.
* Delete the old `Init` implementation.
* Rename `SafeInit` back to `Init` everywhere.

In #6245, I was doing all changes in place in a single PR, which was impossible to reason about.